### PR TITLE
Add option to remove default output in models yaml file

### PR DESCRIPTION
**Pull request by QRemy**
Add option full_output (True by default) to model.to_yaml() and .to_dict(). Switching to False remove the entries for min, max, frozen and error if they are the same than the class defaults or nan.

### CODE REVIEW

File: gammapy/modeling/models/core.py [hunk 1]
```diff
@@ -105,10 +105,20 @@ def copy(self):
...
         return copy.deepcopy(self)
 
-    def to_dict(self):
+    def to_dict(self, full_output=True):
         """Create dict for YAML serialisation"""
         tag = self.tag[0] if isinstance(self.tag, list) else self.tag
-        return {"type": tag, "parameters": self.parameters.to_dict()}
+        params = self.parameters.to_dict()
+
+        if full_output is False:
```
Comment (adonath) [inline gammapy/modeling/models/core.py:113]: Please just use `if not full_output:` here. The `is` pattern kind of works accidentally here...

File: gammapy/modeling/models/core.py [hunk 2]
```diff
@@ -105,10 +105,20 @@ def copy(self):
         """A deep copy."""
         return copy.deepcopy(self)
 
-    def to_dict(self):
+    def to_dict(self, full_output=True):
```
Comment (adonath) [inline gammapy/modeling/models/core.py:108]: I think for writing we should change the default behaviour to `full_output= False`...

**Review by adonath — DISMISSED** [review]
Thanks @QRemy, I have left two minor inline comments...

File: gammapy/modeling/models/core.py (hunk 2 above)
Comment (QRemy) [reply gammapy/modeling/models/core.py:108]: If I change for full_output = False as default then I would not apply simplified output to frozen as it too important to be hidden by default.

Comment (adonath) [reply gammapy/modeling/models/core.py]: I think assuming a parameter is not frozen by default is fine...that's what we do in the `Parameter` class as well.

Comment (QRemy) [reply gammapy/modeling/models/core.py]: ok

**Comment by adonath** [comment]
Thanks @QRemy, the output looks much cleaner now:
```
components:
-   name: pwl-gauss-model
    type: SkyModel
    spectral:
        type: PowerLawSpectralModel
        parameters:
        -   name: index
            value: 2.0
            unit: ''
        -   name: amplitude
            value: 1.0e-12
            unit: cm-2 s-1 TeV-1
        -   name: reference
            value: 1.0
            unit: TeV
    spatial:
        type: GaussianSpatialModel
        frame: icrs
        parameters:
        -   name: lon_0
            value: 0.0
            unit: deg
        -   name: lat_0
            value: 0.0
            unit: deg
        -   name: sigma
            value: 1.0
            unit: deg
        -   name: e
            value: 0.0
            unit: ''
        -   name: phi
            value: 0.0
            unit: deg

```

**Review by adonath — APPROVED** [review]
No further comment from my side...

**Pull request merged**